### ISSUE TITLE
Invalid position given by 'semi'

### ISSUE THREAD

**Issue by dschuan** [AtomLinter/linter-eslint#1312]
ESLint returned a point that did not exist in the document being edited.
Rule: `semi`
Requested start point: 8:3 - 8:5

```js
import React from 'react';

const EmployerManager = () => {
  return (
    <div>
      EmployerManager
    </div>
  )
};

export default EmployerManager;
```

Debug information:
```json
{
  "atomVersion": "1.42.0",
  "linterEslintVersion": "8.5.5",
  "linterEslintConfig": {
    "scopes": [
      "source.js",
      "source.jsx",
      "source.js.jsx",
      "source.flow",
      "source.babel",
      "source.js-semantic"
    ],
    "lintHtmlFiles": false,
    "autofix": {
      "fixOnSave": false,
      "ignoreFixableRulesWhileTyping": false,
      "rulesToDisableWhileFixing": []
    },
    "global": {
      "useGlobalEslint": false,
      "eslintrcPath": "",
      "globalNodePath": ""
    },
    "disabling": {
      "disableWhenNoEslintConfig": true,
      "rulesToSilenceWhileTyping": []
    },
    "advanced": {
      "disableEslintIgnore": false,
      "disableFSCache": false,
      "showRuleIdInMessage": true,
      "eslintRulesDirs": [],
      "localNodeModules": ""
    }
  },
  "eslintVersion": "6.8.0",
  "hoursSinceRestart": 0,
  "platform": "win32",
  "eslintType": "local project",
  "eslintPath": "C:\\Users\\DeSheng\\Documents\\GitHub\\chapati-employers\\node_modules\\eslint",
  "editorScopes": [
    "source.js",
    "keyword.control"
  ]
}
```

**Comment by Arcanemagus** [comment]
What is your ESLint configuration? Can you create a Gist/sample repository with a minimal reproduction case? This looks like JSX code so you must be using a processor of some sort in ESLint.

**Comment by npdev453** [comment]
Problem caused by this eslint update (rollback fix the problem):
https://github.com/eslint/eslint/commit/41a78fd7ce245cad8ff6a96c42f5840688849427

And has 100% repeat with fresh eslint and "semi": enabled rule

**Comment by Arcanemagus** [comment]
Can you please report this as a bug to ESLint then? It looks like the person writing that might have confused 0 and 1 based indexing. I'll try to get some time to verify it with the reproduction from #1315 (since the example provided here is incomplete).

**Comment by mdjermanovic** [comment]
This is probably fixed by eslint/eslint#13275

(#1342 too).
